docs/source/protocol_v3.rst

```diff
@@ -1,51 +1,545 @@
-Proposed Protocol v3
-====================
+IMDv3 Protocol
+==============
 
-The suggested changes to the protocol are as follows:
+IMD (Interactive Molecular Dynamics) is a protocol for communicating molecular simulation 
+data through a socket. 
+It allows for two-way communication: via IMD, a simulation engine sends data to a receiver 
+and the receiver can send forces and certain requests back to the simulation engine.
 
-1. Via an ``.mdp`` file setting, the user should be able to specify which of simulation box, positions, forces, and velocities are sent.
-   These should be rate-adjustable like with xtc and trr output settings but should be set to every step by default.
+Version numbers in IMD are monotonically increasing integers. 
+IMDv3, the protocol version described in this document, builds upon IMDv2, which is implemented 
+at the time of writing in NAMD, VMD, GROMACS, and LAMMPS. IMDv2 is itself 
+a modification of the original IMD protocol published in 2001 :cite:p:`IMDv1:2001`.
 
-2. The IMD_HANDSHAKE packet should have a 1-byte body which contains the configuration settings the simulation was setup with.
-   This will allow the client to be choose appropriate packets to send and receive without redundant configuration.
+.. list-table::
+   :widths: 10 30
+   :header-rows: 1
 
-   The modified handshake packet would look like this:
+   * - IMD version
+     - Protocol specification
+   * - 1
+     - *A system for interactive molecular dynamics simulation* :cite:p:`IMDv1:2001`
+   * - 2
+     - No official specification, but sample protocol and API implementation available :cite:p:`IMDv2`
+   * - 3
+     - This document
+
+.. note:: 
+
+   Because IMDv2 has no official specification, characteristics of the IMDv2 protocol mentioned in this document are inferred from 
+   the sample implementation and commonalities in existing IMDv2 implementations.
+
```

Comment:
Do we have a preferred IMD port in the protocol? If yes, mention it somewhere.

Also mention that the Simulation Engine may freely specify the port at which it listens to a receiver.